### PR TITLE
Adds various Linux distro tests to CI

### PR DESCRIPTION
**Pull request by BradyBonnette**
Originally, this branch was named `fedora-ci`, but as I started developing it I realized that we could be trying out builds in other various distros. With some input from @eeeebbbbrrrr we decided to add as much as we could here for the time being. There is still some stuff we plan on adding in the future, but for now this should cover a lot of use cases.

### CODE REVIEW

**Comment by workingjubilee** [comment]
yay/`_`/yay`_`/RIP/RIP/RIP/yay/RIP/RIP/RIP/RIP/yay/yay/yay//RIP/yay/yay/RIP/RIP/yay/RIP...

**Comment by BradyBonnette** [comment]
@workingjubilee 

Sorry, I wish I could squelch them but trial and error is sometimes the only way as it relates to CI :(

**Comment by workingjubilee** [comment]
You're good.

File: .github/docker/Dockerfile.alpine:3.16
Comment (thomcc) [inline .github/docker/Dockerfile.alpine:3.16:2]: Thank you for documenting this in these!

**Comment by BradyBonnette** [comment]
@eeeebbbbrrrr Note that none of these changes will actually go into effect until this is merged into the default branch according to Github, which for pgx is the `master` branch.

How do you want to proceed? This PR wants to merge into `develop` currently. We could merge it into develop, then:
- Wait until the next release of PGX (where it would end up in `master`), where it would then start working, or
- Cherry pick the squashed commit as a result of this PR that will end up in `develop`, put it in `master` as a one-off, then it will go into effect immediately

**Review by thomcc — APPROVED** [review]
Looks great to me! Thanks for the good documentation too!

File: .github/docker/Dockerfile.alpine:3.16
Comment (BradyBonnette) [reply .github/docker/Dockerfile.alpine:3.16:2]: ❤️ 

I love when I find the same documentation in other places. I figured I'd carry that forward :)

File: .github/docker/Dockerfile.alpine:3.16
Comment (BradyBonnette) [inline .github/docker/Dockerfile.alpine:3.16:4]: 🤦‍♂️ `imagine`

**Comment by eeeebbbbrrrr** [comment]
I think this is fine and good and safe to just merge into master right now.  Lets go ahead and get it running.  I don't mind having some churn on master for CI-based work and issues.

I'll just go ahead and retarget this and merge it now.

**Comment by BradyBonnette** [comment]
Closing this in favor of #822

**Comment by BradyBonnette** [comment]
Also references https://github.com/tcdi/pgx/issues/782

**Pull request closed without merging**
